src/mapeo-manager.js

```diff
@@ -0,0 +1,194 @@
+    /** @type {ProjectKeys} */
+    const keys = {
+      projectKey: projectKeypair.publicKey,
+      projectSecretKey: projectKeypair.secretKey,
+      authEncryptionKey: namespaceEncryptionKeys.auth,
+      blobEncryptionKey: namespaceEncryptionKeys.blob,
+      blobIndexEncryptionKey: namespaceEncryptionKeys.blobIndex,
+      configEncryptionKey: namespaceEncryptionKeys.config,
+      dataEncryptionKey: namespaceEncryptionKeys.data,
+    }
+
+    // TODO: Update to use @mapeo/crypto when ready (https://github.com/digidem/mapeo-core-next/issues/171)
+    const projectId = projectKeypair.publicKey.toString('hex')
+
+    this.#db
+      .insert(projectKeysTable)
+      .values({
+        projectId,
+        encryptionKeys: this.#keyManager.encryptLocalMessage(
+          serialize(keys),
+          projectId
+        ),
+      })
+      .run()
+
+    // 4. Create MapeoProject instance
+    const project = new MapeoProject({
+      projectSettingsConfig: {
+        db: this.#db,
+        indexWriter: this.#projectSettingsIndexWriter,
+      },
+      storagePath: this.#storagePath,
+      encryptionKeys: namespaceEncryptionKeys,
+      keyManager: this.#keyManager,
+      projectKey: projectKeypair.publicKey,
+      projectSecretKey: projectKeypair.secretKey,
+    })
+
+    // 5. Write project name and any other relevant metadata to project instance
+    await project.$setProjectSettings({
+      // TODO: What's the fallback name?
+      name: settings.name || '',
+      defaultPresets: settings.defaultPresets || {},
+    })
+
+    // 6. Return project id
+    return projectId
+  }
+
+  /**
+   * @param {string} projectId
+   * @returns {MapeoProject}
+   */
+  getProject(projectId) {
+    const result = this.#db
+      .select({
+        encryptionKeys: projectKeysTable.encryptionKeys,
+      })
+      .from(projectKeysTable)
+      .where(eq(projectKeysTable.projectId, projectId))
+      .get()
+
+    if (!result.encryptionKeys) {
+      throw new Error(
+        `Could not find encryption keys for project with id ${projectId}`
+      )
+    }
+
+    const keys = deserialize(
+      this.#keyManager.decryptLocalMessage(result.encryptionKeys, projectId)
+    )
+
+    validateProjectKeys(keys)
+
+    const project = new MapeoProject({
+      storagePath: this.#storagePath,
+      projectKey: keys.projectKey,
+      projectSecretKey: keys.projectSecretKey,
+      encryptionKeys: {
+        auth: keys.authEncryptionKey,
+        blob: keys.blobEncryptionKey,
+        blobIndex: keys.blobIndexEncryptionKey,
+        config: keys.configEncryptionKey,
+        data: keys.dataEncryptionKey,
+      },
+      keyManager: this.#keyManager,
+      projectSettingsConfig: {
+        indexWriter: this.#projectSettingsIndexWriter,
+        db: this.#db,
+      },
+    })
+
+    return project
+  }
+}
+
+/**
+ * @param {{[key: string]: unknown}} keys
+ * @returns {asserts keys is ProjectKeys}
+ */
+function validateProjectKeys(keys) {
```

Comment:
open to suggestions for a better way to do this, if possible. basically want to validate the deserialized encryption keys that are read to make sure that they have the correct structure and associated types. Was thinking maybe using Typebox but not sure how it plays with buffer types, since that isn't an official JSON type

Reply:
ok to skip validation here, since there is no unknown input between writes to the DB and reads - same reason drizzle doesn't do validation internally on reads and writes.

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/199/commits/6d14bacb6f178af5102c8d855efffbb0d3b4c085